service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java

```diff
@@ -304,6 +306,70 @@ private void saveParentRelationship(final E entity, final Vertex vertex, final P
         return parents;
     }
 
+    public int getVersion() {
+        try {
+            Vertex versionVertex = getOrCreateVersionVertex();
+            VertexProperty<Integer> property = versionVertex.property(VERSION_PROPERTY_KEY);
+            if (property.isPresent()) {
+                return property.value();
+            } else {
+                throw new IllegalStateException(String.format(
+                        "The version vertex does not conatin the expected property '%s'.", VERSION_PROPERTY_KEY));
+            }
+        } finally {
+            this.graph.tx().commit();
+        }
+
+    }
+
+    public void setVersion(final int version) {
+        try {
+            Vertex versionVertex = getOrCreateVersionVertex();
+            versionVertex.property(VERSION_PROPERTY_KEY, version);
+            this.graph.tx().commit();
+        } catch (Exception e) {
+            this.graph.tx().rollback();
+            throw (e);
+        }
+
+    }
+
+    private Vertex getOrCreateVersionVertex() {
+        try {
+            Vertex versionVertex;
+            // TODO: Composite index created for version property key is not used for the query below.
```

Comment:
This comment is useful. I would remove 'TODO' and remove the composite index, since it is not used.